userspace/libsinsp/test/sinsp.ut.cpp

```diff
@@ -36,3 +37,22 @@ TEST(sinsp, external_event_processor_initialization)
 	EXPECT_EQ(my_sinsp.get_external_event_processor(), &processor);
 }
 
+TEST_F(sinsp_with_test_input, file_open)
+{
+	add_default_init_thread();
+
+	open_inspector();
+	sinsp_evt *evt;
+
+	// since adding and reading events happens on a single thread they can be interleaved.
+	// tests may need to change if that will not be the case anymore
+	add_event(increasing_ts(), 1, PPME_SYSCALL_OPEN_E, 3, "/tmp/the_file", PPM_O_RDWR, 0);
+	evt = next_event();
+
+	add_event(increasing_ts(), 1, PPME_SYSCALL_OPEN_X, 6, 3, "/tmp/the_file", PPM_O_RDWR, 0, 5, 123);
```

Comment:
Looking at the code; is the `next_event` always needed (or, well, most of the time) after the `add_event` call?
If so, can't `add_event` take a `sinsp_evt*` argument and, when not-nil, nex_event gets automatically called?

Reply:
The two approaches are very similar, but I prefer it to be like this to be a bit more flexible:
* currently (like the comment says) you _can_ interleave add_event with next_event, but if that stops being the case you will be required to add all the add_event before the test starts and next_event after.
* you might want to do something fancier when generating events (such as having for loops that generate the same operations for different processes or something like that) and then have a different logic when checking them. Leaving the two operations separate gives you the maximum flexibility I think.
* Another thing we might add if we see that adding `next_event()` is too much of a hassle could be a function that consumes all events that the test generated so far.
* The idea about an optional parameter is cool but IMO it might look a little confusing to a new contributor that wants to add a test

Reply:
I have no strong opinion in that sense :) it was just a suggestion i came up with while reading through the code.
:+1: from me!

Reply:
We can do something about it later in any case!

Reply:
Agree with @LucaGuerra here, BTW we can always add something like this if needed